src/components/Todo.tsx

```diff
@@ -40,18 +55,39 @@ function Todo(props: { fetchList: any; addTodo: any }) {
     event.preventDefault();
     try {
       const response = await props.addTodo(todo);
-      setNewdata(response["task"]);
+      setNewdata(response);
       setWorking(true);
     } catch {
       setWorking(false);
     }
     setTodo("");
   };
 
+  const [del, setDel] = useState(false);
+  const [id, setId] = useState(0);
+  const deleteItem = async (event: any) => {
+    const buttonId = event.currentTarget.getAttribute("data-testid");
+    setId(buttonId);
+    const canDelete = await deleteTodo(buttonId);
+    setDel(canDelete);
```

Comment:
Is the idea that the delete button goes away if delete fails? I'm a little confused here.

Reply:
No, the delete button doesn't go away. The idea is that I run the DELETE method and if I get a `response.ok = True` then the `setDel` will be true and that sets off a `useEffect` which removes the deleted item from the list. If the DELETE method returns a `response.ok=False` then nothing will happen because the `setDel` is initially set to `False` so if it doesn't change, then the `useEffect` won't run. I hope this makes sense.